### PR TITLE
move broker redirecturi into common which can be reused later in MSAL…

### PR DESCRIPTION
**Pull request by kaisong1990**
… cpp

## Proposed changes

Moving some code from MSALRedirectUri by adding a new category on NSURL. The main goal here is we can reuse the code from common from CPP side

Describe what this PR is trying to do.

## Type of change

- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [x] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

**Comment by kaisong1990** [comment]
The branch name might be confusing. It should be redirectUrl instead of requestUri

File: IdentityCore/src/util/NSURL+MSIDBrokerRedirectUri.m [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+@implementation NSURL (MSIDBrokerRedirectUri)
+
++ (NSURL *)defaultNonBrokerRedirectUri:(NSString *)clientId
+{
+    if ([NSString msidIsStringNilOrBlank:clientId])
+    {
+        return nil;
+    }
+    
+    NSString *redirectUri = [NSString stringWithFormat:@"msal%@://auth", clientId];
```
Comment (oldalton) [inline IdentityCore/src/util/NSURL+MSIDBrokerRedirectUri.m:37]: Can we check telemetry to see if this one is still used by anyone? Maybe it is a good time to drop support for this one :)

File: IdentityCore/tests/MSIDBrokerRedirectUriValidationTests.m [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+#import <XCTest/XCTest.h>
+#import "NSURL+MSIDBrokerRedirectUri.h"
+
+@interface MSIDBrokerRedirectUriValidationTests : XCTestCase
+
+@end
+
+@implementation MSIDBrokerRedirectUriValidationTests
+
+- (void)test_default_non_broker_redirectUri_with_nil_or_empty_cliendid
```
Comment (oldalton) [inline IdentityCore/tests/MSIDBrokerRedirectUriValidationTests.m:35]: I believe there were other tests in MSAL for this. Let's move them as well to common core and remove from MSAL.

**Comment by kaisong1990** [comment]
> What about scheme registration check logic? I believe it will be needed for CPP as well. I feel like we need to move entire MSALRedirectUriVerifier logic here for both iOS and macOS to support end to end flows.

Yes, I planned to move that in as a separate PR and keep this PR smaller

File: IdentityCore/src/util/NSURL+MSIDBrokerRedirectUri.m (hunk 1 above)
Comment (kaisong1990) [reply IdentityCore/src/util/NSURL+MSIDBrokerRedirectUri.m:37]: Would you provide more guidance on how to check this one? I didn't see there is telemetry around the caller.

Comment (oldalton) [reply IdentityCore/src/util/NSURL+MSIDBrokerRedirectUri.m:37]: I believe you can look it up from ESTS telemetry (e.g. check MaskedResponse field)

**Comment by kaisong1990** [comment]
@oldalton Closed this PR as the branch name is confusing and there is a typo in it. Here is the new PR of moving in redirect uri and its verifier including utests:

https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/pull/807

**Pull request closed without merging**